folium/colormap.py

```diff
@@ -0,0 +1,594 @@
+                        '#e6f598', '#99d594', '#3288bd'],
+           'Accent': ['#7fc97f', '#beaed4', '#fdc086',
+                      '#ffff99', '#386cb0', '#f0027f'],
+           'Dark2': ['#1b9e77', '#d95f02', '#7570b3',
+                     '#e7298a', '#66a61e', '#e6ab02'],
+           'Paired': ['#a6cee3', '#1f78b4', '#b2df8a',
+                      '#33a02c', '#fb9a99', '#e31a1c'],
+           'Pastel1': ['#fbb4ae', '#b3cde3', '#ccebc5',
+                       '#decbe4', '#fed9a6', '#ffffcc'],
+           'Pastel2': ['#b3e2cd', '#fdcdac', '#cbd5e8',
+                       '#f4cae4', '#e6f5c9', '#fff2ae'],
+           'Set1': ['#e41a1c', '#377eb8', '#4daf4a',
+                    '#984ea3', '#ff7f00', '#ffff33'],
+           'Set2': ['#66c2a5', '#fc8d62', '#8da0cb',
+                    '#e78ac3', '#a6d854', '#ffd92f'],
+           'Set3': ['#8dd3c7', '#ffffb3', '#bebada',
+                    '#fb8072', '#80b1d3', '#fdb462'],
+          }
+
+def _parse_color(x):
+    """
+    """
+    if isinstance(x, tuple) or isinstance(x, list):
+        color_tuple = tuple(x)[:4]
+    elif isinstance(x, text_type) or isinstance(x, binary_type):
+        if x[0]=="#" and len(x)==7:
+            # Color of the for #RRGGBB
+            color_tuple = (int(x[1:3],16), int(x[3:5],16), int(x[5:7],16))
+        else:
+            color_code = _cnames.get(x.lower(),None)
+            if color_code is None:
+                raise ValueError('Unknown color.')
+            color_tuple = (int(color_code[1:3],16), int(color_code[3:5],16), int(color_code[5:7],16))
+    if max(color_tuple)>1.:
+        color_tuple = tuple(u/255. for u in color_tuple)
+    return tuple(map(float,(color_tuple+(1.,))[:4]))
+
+_round = round
+
+def _base(x):
+    if x > 0:
+        base = pow(10, math.floor(math.log10(x)))
+        return round(x/base)*base
+    else:
+        return 0
+
+class ColorMap(MacroElement):
+    """A generic class for creating colormaps."""
+
+    def __init__(self, min=0., max=1., caption=""):
+        """
+        """
+        super(ColorMap, self).__init__()
+        self._name = 'ColorMap'
+
+        self.min = min
+        self.max = max
+        self.caption = caption
+        self.index = [min, max]
+
+        self._template = self._env.get_template('color_scale.js')
+
+    def render(self, **kwargs):
+        self.color_domain = [self.min + (self.max-self.min)*i/499. for i in range(500)]
+        self.color_range = [self.__call__(x) for x in self.color_domain]
+        self.tick_labels = legend_scaler(self.index)
+        self.caption = ""
+
+        super(ColorMap, self).render(**kwargs)
+
+        figure = self.get_root()
+        assert isinstance(figure, Figure), ("You cannot render this Element "
+                                            "if it's not in a Figure.")
+
+        figure.header.add_children(
+            JavascriptLink("https://cdnjs.cloudflare.com/ajax/libs/d3/3.5.5/d3.min.js"),  # noqa
+            name='d3')
```

Comment:
if you are using `#noqa` you can make this 1-line.  It those moments where pep8 hurts readability... :arrow_down_small: :ghost: (Could not find an upside down ghost emoji.)